pkg/provisioningv2/rke2/planner/certificaterotation.go

```diff
@@ -26,9 +28,9 @@ func (r *certificateRotation) RotateCertificates(controlPlane *rkev1.RKEControlP
 		return nil
 	}
 
-	rotatePlan := rotateCertificatesPlan(controlPlane, controlPlane.Spec.RotateCertificates)
-	for _, node := range collect(clusterPlan, isControlPlaneEtcd) {
-		err := r.store.UpdatePlan(node, rotatePlan, 0)
+	for _, node := range collect(clusterPlan, anyRole) {
+		rotatePlan := rotateCertificatesPlan(controlPlane, controlPlane.Spec.RotateCertificates, node)
+		err := assignAndCheckPlan(r.store, fmt.Sprintf("[%s] certificate rotation", node.Machine.Name), node, rotatePlan, 0)
```

Comment:
The only concern I think there can be with this approach is that `assignAndCheckPlan` is a blocking call, and thus certificate rotation will happen one-at-a-time per node. We really need a `allPlansInSync` type of blocking call that can be used... but not blocking

Reply:
Not sure if it changed with your pr but for me this still worked concurrently because the function returns the sync error and when it gets queued up again it fires a different node plan off so it still runs concurrently it is just not setting all the nodes in 1 go per controller call.